src/include/storage/fd.h

```diff
@@ -145,4 +145,6 @@ extern const char *FileGetFilename(File file);
 extern void FileSetIsWorkfile(File file);
 extern void FileSetIsTempFile(File file, bool isTempFile);
 
+extern int	TruncateFileByName(FileName fileName);
```

Comment:
Why not simply put public do_truncate to md.h like mdunlink?

Reply:
Now functions in md.c refer to aomd.h and fd.h.
And functions in aomd.c refer to fd.h.

If we just put do_truncate to md.h, than we will introduce a new link - when functions from aomd component refer to md component.
That will increase coupling.

Moving functionality to fd.c level seems better solution to keep current coupling level.

Reply:
> functions from aomd component refer to md component

Isn't this logical? aomd is md for AO

Reply:
Function that truncates file basing on its name is more suitable for low level api, like fd.h level. 
It is an utility function related to single file handling, not to storage manager.

But I would agree that there is some benefit in leaving it in md.c - it will introduce less delta comparing with usual gpdb.

Reply:
> benefit in leaving it in md.c - it will introduce less delta comparing with usual gpdb.

yes

Reply:
Ok, discussed with Andrey Sokolov - will return do_truncate to md.c to make less delta with gpdb.